Kconfig.sof

```diff
@@ -207,6 +207,13 @@ config DSP_RESIDENCY_COUNTERS
 	  R0, R1, R2 are abstract states which can be used differently
 	  based on platform implementation.
 
+config MODULES
+	bool "Enable modules"
```

Comment:
I think that it is not explanatory enough. Should it be native loadable modules?

Reply:
@pjdobrowolski sure, good idea, thanks, updated

Reply:
Yeah, we need to really namespace all our KConfig with ```SOF_``` prefix too, to avoid collisions with Zephyr as we share lots of common names. We are not compliant today.

Reply:
@lgirdwood I think some tooling actually requires `CONFIG_MODULES` to allow the use of `tristate` in Kconfig options. But yes, maybe we need this in Zephyr, I was thinking about that too

Reply:
...just double-checked: `config MODULES` is indeed needed for kconfiglib, otherwise selecting "m" for `tristate` options doesn't work

Reply:
I pushed https://github.com/zephyrproject-rtos/zephyr/pull/67105 - if it gets accepted in Zephyr, we'll drop it here